packages/suite/src/components/suite/modals/ReduxModal/UserContextModal/AddAccountModal/AddAccountModal.tsx

```diff
@@ -108,25 +113,29 @@ export const AddAccountModal = ({ device, onCancel, symbol, noRedirect }: AddAcc
 
     // Collect possible accountTypes
     const accountTypes = isSelectedNetworkEnabled
-        ? networksCompatibility
-              .filter(n => n.symbol === selectedNetwork.symbol)
+        ? (pipe(
+              selectedNetwork.accountTypes,
+              accountTypesToArray,
               /**
                * Filter out coinjoin account type if it is not visible.
                * Visibility of coinjoin account type depends on coinjoin feature config in message system.
                * By default it is visible publicly, but it can be remotely hidden under debug menu.
                */
-              .filter(({ backendType }) => backendType !== 'coinjoin' || isCoinjoinVisible)
-              .filter(
-                  ({ isDebugOnlyAccountType }) =>
-                      isDebugOnlyAccountType === undefined || (isDebugOnlyAccountType && isDebug),
-              )
+              A.filter(({ backendType }) => backendType !== 'coinjoin' || isCoinjoinVisible),
+              A.filter(({ isDebugOnlyAccountType }) => !isDebugOnlyAccountType || isDebug),
+              // the default account is expected to be the first one
+              A.prepend({
+                  bip43Path: selectedNetwork.bip43Path,
+                  accountType: 'normal',
+              } as NetworkAccountCollection[number]),
+          ) as NetworkAccountCollection)
```

Comment:
This is a good usecase for functional programming, but unfortunately `mobily-belt A` is poorly typed. It converts all return types to `readonly`, so I have to use `as` to make it mutable again, otherwise  types are incompatible :facepalm: 

I'm kind of disappointed with `mobily-belt` compared to `ramda` or `fp-ts`, and maybe I should just use vanilla filter and unshift. What do you think? :thinking:

Reply:
I get what you're saying - that `reduce` written this way restricts the type, which could be useful. Though in this case, restricting type does not help; the code that consumes `accountTypes` does not look at those properties so it'd make no difference.

But it's a nice pattern. Though if we needed this same could be done with `filter` (imho more intuitive to to read):
```
const arr2: A[] = arr.filter((item): item is A => item === "a")
```

Reply:
https://github.com/trezor/trezor-suite/pull/14109/commits/16e8c791d26bc22e0f22a854850f64f8bf99549b